### DISCUSSION TITLE
Obtaining variable Information

### DISCUSSION THREAD

**Discussion by MikeGrateful** [mozilla/rhino#1282] [Q&A]
![image](https://user-images.githubusercontent.com/15280143/201459667-31a6b25c-b961-4b5b-92b1-8c902e719da6.png)
May I ask how to obtain the variable table and stack information in the onDebuggerStatement method? I have tried many methods but failed. The demo debug program has no relevant functional code

**Comment by p-bakker** [comment]
Please stop creating issues that aren't issues

**Comment by p-bakker** [comment]
Based on https://p-bakker.github.io/rhino/tools/debugger/#viewing-variables I conclude that the Debugger implementation that is part of Rhino is able to show variables, so the code must give you some clues as to how its done

You might wanna take a look at the dbgp debugger implementation mentioned in https://github.com/mozilla/rhino/issues/973#issue-933454026, which I know for sure also displays variables

**Comment by MikeGrateful** [comment]
When I'm trying to get an interpreter to suspend the stack, variables, and other related information.
 https://p-bakker.github.io/rhino/tools/debugger/#viewing-variables the debug seem to have no useful information
